landing-page/content/common/how-to-release.md

```diff
@@ -263,64 +292,102 @@ Java artifacts are available from Maven Central.
+#### Codebase updates
+
+There are a few places in the codebase that reference the latest Iceberg release version number and need to be updated:
+1. https://github.com/apache/iceberg/blob/master/.github/ISSUE_TEMPLATE/iceberg_bug_report.yml
+2. https://github.com/apache/iceberg/blob/master/python/dev/Dockerfile
+3. `oldVersion` in `revapi` in https://github.com/apache/iceberg/blob/master/build.gradle
+
+### Documentation release
 
 Documentation needs to be updated as a part of an Iceberg release after a release candidate is passed.
-The commands described below assume you are in a directory containing a local clone of the `iceberg-docs`
-repository and `iceberg` repository. Adjust the commands accordingly if it is not the case. Note that all
-changes in `iceberg` need to happen against the `master` branch and changes in `iceberg-docs` need to happen
-against the `main` branch. 
 
-#### iceberg repository preparations
+#### Prerequisites
+
+Similar to the `iceberg` repository, you also need to set up `https://github.com/apache/iceberg-docs.git` as a remote with name `apache`.
 
-A PR needs to be published in the `iceberg` repository with the following changes:
+The commands described below assume you have a directory structure of:
 
-1. Create a new folder called `docs/releases/<VERSION NUMBER>` with an `_index.md` file. See the existing folders under `docs/releases` for more details.
+```text
+/some/path
+├── iceberg
+└── iceberg-docs
+```
+
+And you are:
+1. in the `iceberg-docs` repository.
+2. have `iceberg` repository on the specific version branch
 
-#### Common documentation update
+Adjust the commands below accordingly if it is not the case.
+
+#### Update specs
+
+Copy the latest format specifications to `landing-page/content/common`:
 
-1. To start the release process, run the following steps in the `iceberg-docs` repository to copy docs over:
 ```shell
 cp -r ../iceberg/format/* ../iceberg-docs/landing-page/content/common/
 ```
-2. Change into the `iceberg-docs` repository and create a branch.
-```shell
-cd ../iceberg-docs
-git checkout -b <BRANCH NAME>
-```
-3. Commit, push, and open a PR against the `iceberg-docs` repo (`<BRANCH NAME>` -> `main`)
 
-#### Versioned documentation update
+Raise a PR with the specific changes against `main` branch and merge.
 
-Once the common docs changes have been merged into `main`, the next step is to update the versioned docs.
+#### Copy versioned documentations
+
+Copy the versioned docs into `docs/content`
 
-1. In the `iceberg-docs` repository, cut a new branch using the version number as the branch name
-```shell
-cd ../iceberg-docs
-git checkout -b <VERSION>
-git push --set-upstream apache <VERSION>
-```
-2. Copy the versioned docs from the `iceberg` repo into the `iceberg-docs` repo
 ```shell
 rm -rf ../iceberg-docs/docs/content
 cp -r ../iceberg/docs ../iceberg-docs/docs/content
 ```
-3. Commit the changes and open a PR against the `<VERSION>` branch in the `iceberg-docs` repo
```

Comment:
In the new workflow described here, we do everything against `main`, and in the end cut the version branch and sync latest branch, this avoids unnecessary commit conflicts and need to do branch merges.